test/system/scenarios/test_electrodes.py

```diff
@@ -426,6 +426,119 @@ def issue497(sim):
     assert_true (abs(i_amp_ac2[int(start2/sim_dt)]-amplitude) < 1e-9)
 
 
+@register()
+def issue512(sim):
+    """
+    Test to ensure that StepCurrentSource times are handled similarly across
+    all simulators. Multiple combinations of step times tested for:
+    1) dt = 0.1 ms, min_delay = 0.1 ms
+    2) dt = 0.01 ms, min_delay = 0.01 ms
+    Note: exact matches of times not appropriate owing to floating point
+    rounding errors. If absolute difference <1e-9, then considered equal.
+    """
+    def get_len(data):
+        if "pyNN.nest" in str(sim):
+            # as NEST uses LazyArray
+            return len(data.evaluate())
+        else:
+            return len(data)
+
+    # 1) dt = 0.1 ms, min_delay = 0.1 ms
+    dt = 0.1
+    sim.setup(timestep=dt, min_delay=dt)
+    cells = sim.Population(1, sim.IF_curr_exp(v_thresh=-55.0, tau_refrac=5.0, v_rest=-60.0))
+    # 1.1) Negative time value
+    flag = 0
+    try:
+        step = sim.StepCurrentSource(times=[0.4, -0.6, 0.8],
+                                     amplitudes=[0.5, -0.5, 0.5])
+    except ValueError:
+        flag = 1
+    assert_true (flag == 1)
```

Comment:
`using assert_equal(flag, 1)` will give more helpful error messages when it fails 

(and similarly for most uses of `assert_true` below)

Reply:
Although in fact, what you should use here is
`assert_raises(ValueError, sim.StepCurrentSource, times=[0.4, -0.6, 0.8], amplitudes=[0.5, -0.5, 0.5])`

Reply:
That's useful. Nice to learn that. Thanks!